Automattic/wp-calypso
# Self-service site restore: update UI

Related to https://github.com/Automattic/dotcom-forge/issues/6287 
Related to D143388-code
Related to D143193-code

<img width="216" alt="Screenshot 2567-03-28 at 12 44 08" src="https://github.com/Automattic/wp-calypso/assets/6851384/4cd734fe-e919-4115-b1c0-3e025be2887c">


<img width="1116" alt="Screenshot 2567-04-01 at 17 41 52" src="https://github.com/Automattic/wp-calypso/assets/6851384/8253565a-8f3c-4cdd-a451-d7a9de9e6707">

<img width="388" alt="Screenshot 2567-04-01 at 17 41 39" src="https://github.com/Automattic/wp-calypso/assets/6851384/30f2400b-8314-4c95-b31a-49566365e78d">


## Proposed Changes

* Allow accessing and restoring deleted sites at `/sites` when the status **Deleted** is selected. Deleted sites are not returned for any other status. 

## Testing Instructions



* Delete a site
* Go to /sites
* Filter to "Deleted"
* Restore the site
* Filter to "All"
* See the site is back and that you're admin 
* Try it with multiple users. They all should be added back. 

## Pre-merge Checklist
